Latest main fails to build

I tried rebasing my branch locally and now my builds fail with:
```cs
H:\Projects\dotnet\runtime\src\libraries\System.Text.Json\gen\JsonSourceGenerator.Parser.cs(284,22): error IDE0062: Local function can be made static (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0062) [H:\Projects\dotnet\runtime\src\libraries\System.Text.Json\gen\System.Text.Json.SourceGeneration.Roslyn4.4.csproj::TargetFramework=netstandard2.0]
H:\Projects\dotnet\runtime\src\libraries\System.Text.Json\gen\JsonSourceGenerator.Parser.cs(284,22): error IDE0062: Local function can be made static (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0062) [H:\Projects\dotnet\runtime\src\libraries\System.Text.Json\gen\System.Text.Json.SourceGeneration.Roslyn3.11.csproj]
H:\Projects\dotnet\runtime\src\libraries\System.Text.Json\gen\JsonSourceGenerator.Parser.cs(284,22): error IDE0062: Local function can be made static (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0062) [H:\Projects\dotnet\runtime\src\libraries\System.Text.Json\gen\System.Text.Json.SourceGeneration.Roslyn4.0.csproj]
```

--- a/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs
+++ b/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs
@@ -281,7 +281,7 @@ namespace System.Text.Json.SourceGeneration
                     }
                 }
 
-                void AddTypeArgumentDiagnosticIds(ITypeSymbol type, ref HashSet<string>? experimentalIds)
+                static void AddTypeArgumentDiagnosticIds(ITypeSymbol type, ref HashSet<string>? experimentalIds)
                 {
                     if (type is IArrayTypeSymbol arrayType)
                     {

